docs/proposals/use-cascading-deletion/README.md

```diff
@@ -0,0 +1,137 @@
+
+#### Story 2
+<!--
+作为管理员, 我希望可以有一种手段可以将某个测试集群从Karmada控制面移除,同时测试集群上面的workload不受影响
+-->
+As an administrator, I hope there is a way to remove a test cluster from the Karmada control plane without affecting the workload on the test cluster.
+
+#### Story 3
+<!--
+作为管理员,为了降低接口耗时,我希望在删除控制面资源模板之后可以立即返回,不阻塞等待直到成员集群资源删除。
+-->
+As an administrator, in order to reduce the interface time-consuming, I hope that I can return immediately after deleting the control plane resource template without blocking and waiting until the member cluster resources are deleted.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+### API changes
+<!--
+- 由于删除操作的对象是资源模板, 所以为资源模板添加Annotation, 名称为: `resourcetemplate.karmada.io/cascadedeletion`
+- 为了可扩展性,value应该是一个枚举类型
+  - orphan: 删除work对象,清理Karmada附加给成员集群的workload的annotation/labels,不删除成员集群workload对象.
+  - foreground: 默认情况前台删除,等同于不存在上述注解或者注解的值为空;删除动作会阻塞等待直到成员集群workload被删除掉.
+  - background: 后台删除表示删除资源模板之后立即返回,由控制器在后台逐步清理成员集群workload.
+-->
+
+- Since the object of the deletion operation is the resource template, add Annotation to the resource template, named: `resourcetemplate.karmada.io/cascadedeletion`
+- For scalability, value should be an enum type
+    - orphan: Delete the work object, clean up the annotation/labels of the workload attached by Karmada to the member cluster, and do not delete the member cluster workload object.
+    - foreground: By default, the foreground is deleted, which means that the above annotation does not exist or the value of the annotation is empty; the deletion action will block and wait until the member cluster workload is deleted.
+    - background: Background deletion means returning immediately after deleting the resource template, and the controller gradually cleans up the member cluster workload in the background.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    propagationpolicy.karmada.io/name: foo
+    propagationpolicy.karmada.io/namespace: default
+    resourcetemplate.karmada.io/cascadedeletion: orphan
+```
+
+<!--
+- 用户触发删除动作
+- execution-controller 根据级联删除策略执行相应的动作
+  ![Dependencies resource deletion policy](statics/dependencies-resource-deletion-policy.png)
+-->
+
+- User triggers deletion action
+- execution-controller performs corresponding actions according to the cascade deletion policy
+  ![use-cascading-deletion](statics/use-cascading-deletion.png)
+
+### The cascading deletion policy of dependent resources and main resources does not force binding
+<!--
+依赖资源和主资源的级联删除策略不强制绑定
+由于依赖资源可能被多个资源模版共享,在这种情况下很难决策依赖资源的删除策略以哪个删除策略为准; 不强制和主资源绑定,由用户自己决策,灵活性和扩展性更好
+-->
+Since dependent resources may be shared by multiple resource templates, in this case it is difficult to decide which deletion strategy should be used for the dependent resources; it is not forced to be bound to the main resource, and is left to the user to decide, with greater flexibility and scalability. good
```

Comment:
This could also easily lead to such a situation where the deletion policy for the following resources (such as ConfigMap) is set to cascade deletion, but the deletion policy for the leading resources (such as Deployment) is set to retention. When users delete the following resources, it causes the applications in the member clusters to become abnormal.